core/src/main/scala/jms4s/JmsAcknowledgerConsumer.scala

```diff
@@ -119,35 +46,30 @@ object JmsAcknowledgerConsumer {
         .as(
           Stream.eval(
             for {
-              resource <- pool.dequeue1
-              message  <- resource.consumer.receiveJmsMessage
-              res      <- f(message)
+              (context, consumer) <- pool.dequeue1
+              message             <- consumer.receiveJmsMessage
+              res                 <- f(message)
               _ <- res.fold(
                     ifAck = blocker.delay(message.wrapped.acknowledge()),
                     ifNoAck = Sync[F].unit,
                     ifSend = send =>
                       blocker.blockOn(
```

Comment:
I noticed this `blockOn` will block too much..
The following inner `context.send`s will block as well, so we can do better.